dsl/src/main/scala/com.crobox.clickhouse/dsl/Query.scala

```diff
@@ -10,6 +10,19 @@ trait Table {
   val columns: List[NativeColumn[_]]
 }
 
+/**
+ * An extension on Table for case classes, 
+ * that provides an implementation of the column list for its own properties.
+ */
+trait CTable extends Table { self: Product =>
```

Comment:
Should we add a simple unit test for this?

Reply:
I think we should, ill add it when I find some time.